{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <p style=\"text-align: right;\"> &#9989; **Context Matters** </p>\n",
    "#### <p style=\"text-align: right;\"> &#9989; Christina, Hao, Yunfei, Erik</p>\n",
    "\n",
    "# Module 3\n",
    "\n",
    "## How do gene transcription relate to phenotype?\n",
    "\n",
    "We will work with maize.\n",
    "\n",
    "### Preliminaries\n",
    "\n",
    "First, import the usual libraries\n",
    "- `math`: basic math operations\n",
    "- `os`: enable file manipulation with the OS\n",
    "- `sys`: enable interaction with commandline\n",
    "- `importlib` : reload libraries if necessary\n",
    "- `argparse` : pass arguments directly to the command line\n",
    "- `glob`: more variable manipulation\n",
    "- `matplotlib.pyplot`: default plotter (I personally like ggplot waaaaay better. E)\n",
    "    - `inline`: so that plots are shown in the notebook\n",
    "- `numpy`: all number cruching done here\n",
    "- `pandas`: data wrangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "import importlib\n",
    "import numpy as np\n",
    "import os\n",
    "import sys\n",
    "import argparse\n",
    "import glob\n",
    "from matplotlib import pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "%matplotlib inline\n",
    "from mpl_toolkits.mplot3d import Axes3D\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "sns.set(color_codes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Specify the gene expression raw data file and a directory to save plots and CSVs.\n",
    "\n",
    "The eQTL file was provided by (Christina,Fabio,Nolan,Scott)-group in Module 1. Supposedly, it is the result of eQTL performed of every gene against every SNP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path to FPKM data\n",
    "gene_file = '~/documents/css893/942_FPKM_B73_genes_w_feature.txt'\n",
    "\n",
    "# path to eQTL data\n",
    "eqtl_file = '/home/ejam/documents/css893/Merged_eQTL_Data.tsv'\n",
    "\n",
    "# path to correlation matrix directory\n",
    "src = '/home/ejam/documents/css893/tpm_corr/'\n",
    "\n",
    "# path to save results\n",
    "dst = '/home/ejam/documents/css893/context_matters/results/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the original FPKM data. This time we only care of the metadata. All the contigs were treated as _chromosome 11_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ejam/.local/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "fpkm = pd.read_table(gene_file)\n",
    "metadata_cols = 5\n",
    "\n",
    "gene_info = fpkm.iloc[:, :metadata_cols]\n",
    "foo = gene_info['chromosome'].unique()\n",
    "\n",
    "for idx in foo:\n",
    "    if idx == str(idx):\n",
    "        value = 0\n",
    "        if idx[0] == 'B':\n",
    "            value = int('73' + (idx.split('_')[-1])[3:])\n",
    "            value = 11\n",
    "        else:\n",
    "            value = int(idx)\n",
    "        \n",
    "        mask = gene_info.chromosome == idx\n",
    "        gene_info.loc[mask, 'chromosome'] = value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the correlation matrix. The correlation method is _pearson_ and we only consider those genes whose TPM variance was above 10e5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>60</th>\n",
       "      <th>74</th>\n",
       "      <th>85</th>\n",
       "      <th>95</th>\n",
       "      <th>114</th>\n",
       "      <th>169</th>\n",
       "      <th>199</th>\n",
       "      <th>415</th>\n",
       "      <th>562</th>\n",
       "      <th>657</th>\n",
       "      <th>...</th>\n",
       "      <th>43720</th>\n",
       "      <th>43830</th>\n",
       "      <th>43873</th>\n",
       "      <th>44001</th>\n",
       "      <th>44033</th>\n",
       "      <th>44111</th>\n",
       "      <th>44119</th>\n",
       "      <th>44155</th>\n",
       "      <th>44221</th>\n",
       "      <th>44254</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.203803</td>\n",
       "      <td>-0.041840</td>\n",
       "      <td>0.135602</td>\n",
       "      <td>0.297998</td>\n",
       "      <td>-0.025113</td>\n",
       "      <td>-0.373148</td>\n",
       "      <td>0.335764</td>\n",
       "      <td>-0.295790</td>\n",
       "      <td>0.397197</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.309133</td>\n",
       "      <td>-0.326422</td>\n",
       "      <td>-0.096837</td>\n",
       "      <td>-0.186167</td>\n",
       "      <td>0.136122</td>\n",
       "      <td>-0.417258</td>\n",
       "      <td>0.143673</td>\n",
       "      <td>-0.036973</td>\n",
       "      <td>0.028265</td>\n",
       "      <td>-0.382432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.203803</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.084412</td>\n",
       "      <td>-0.320546</td>\n",
       "      <td>-0.071207</td>\n",
       "      <td>0.101266</td>\n",
       "      <td>0.424646</td>\n",
       "      <td>-0.175569</td>\n",
       "      <td>0.446181</td>\n",
       "      <td>-0.486499</td>\n",
       "      <td>...</td>\n",
       "      <td>0.386614</td>\n",
       "      <td>0.424727</td>\n",
       "      <td>0.153691</td>\n",
       "      <td>0.069929</td>\n",
       "      <td>-0.348278</td>\n",
       "      <td>0.370390</td>\n",
       "      <td>0.104439</td>\n",
       "      <td>-0.068258</td>\n",
       "      <td>0.202899</td>\n",
       "      <td>0.455165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-0.041840</td>\n",
       "      <td>0.084412</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.045254</td>\n",
       "      <td>-0.170664</td>\n",
       "      <td>-0.124421</td>\n",
       "      <td>0.323632</td>\n",
       "      <td>-0.062687</td>\n",
       "      <td>0.161443</td>\n",
       "      <td>-0.193569</td>\n",
       "      <td>...</td>\n",
       "      <td>0.360204</td>\n",
       "      <td>0.271762</td>\n",
       "      <td>0.036250</td>\n",
       "      <td>-0.038425</td>\n",
       "      <td>0.057585</td>\n",
       "      <td>0.115981</td>\n",
       "      <td>0.112563</td>\n",
       "      <td>0.080347</td>\n",
       "      <td>0.111763</td>\n",
       "      <td>0.141706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.135602</td>\n",
       "      <td>-0.320546</td>\n",
       "      <td>-0.045254</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.097013</td>\n",
       "      <td>0.056162</td>\n",
       "      <td>-0.387976</td>\n",
       "      <td>0.232699</td>\n",
       "      <td>-0.433461</td>\n",
       "      <td>0.467508</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.401847</td>\n",
       "      <td>-0.356511</td>\n",
       "      <td>-0.150706</td>\n",
       "      <td>-0.066169</td>\n",
       "      <td>0.363566</td>\n",
       "      <td>-0.399549</td>\n",
       "      <td>-0.029409</td>\n",
       "      <td>0.056737</td>\n",
       "      <td>-0.233748</td>\n",
       "      <td>-0.494055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.297998</td>\n",
       "      <td>-0.071207</td>\n",
       "      <td>-0.170664</td>\n",
       "      <td>0.097013</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.076526</td>\n",
       "      <td>-0.309533</td>\n",
       "      <td>0.360927</td>\n",
       "      <td>-0.220001</td>\n",
       "      <td>0.279083</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.307539</td>\n",
       "      <td>-0.251303</td>\n",
       "      <td>0.034189</td>\n",
       "      <td>-0.165472</td>\n",
       "      <td>-0.075354</td>\n",
       "      <td>-0.314652</td>\n",
       "      <td>0.182568</td>\n",
       "      <td>-0.121469</td>\n",
       "      <td>0.038715</td>\n",
       "      <td>-0.243578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>643</th>\n",
       "      <td>-0.417258</td>\n",
       "      <td>0.370390</td>\n",
       "      <td>0.115981</td>\n",
       "      <td>-0.399549</td>\n",
       "      <td>-0.314652</td>\n",
       "      <td>0.030274</td>\n",
       "      <td>0.669776</td>\n",
       "      <td>-0.405258</td>\n",
       "      <td>0.643508</td>\n",
       "      <td>-0.785690</td>\n",
       "      <td>...</td>\n",
       "      <td>0.605081</td>\n",
       "      <td>0.549997</td>\n",
       "      <td>0.256985</td>\n",
       "      <td>0.361084</td>\n",
       "      <td>-0.449045</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.084839</td>\n",
       "      <td>-0.007476</td>\n",
       "      <td>0.207629</td>\n",
       "      <td>0.803752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>644</th>\n",
       "      <td>0.143673</td>\n",
       "      <td>0.104439</td>\n",
       "      <td>0.112563</td>\n",
       "      <td>-0.029409</td>\n",
       "      <td>0.182568</td>\n",
       "      <td>0.087487</td>\n",
       "      <td>0.010110</td>\n",
       "      <td>0.133939</td>\n",
       "      <td>0.118709</td>\n",
       "      <td>-0.055325</td>\n",
       "      <td>...</td>\n",
       "      <td>0.083770</td>\n",
       "      <td>-0.110514</td>\n",
       "      <td>0.022980</td>\n",
       "      <td>-0.062397</td>\n",
       "      <td>-0.051659</td>\n",
       "      <td>-0.084839</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.097842</td>\n",
       "      <td>0.183584</td>\n",
       "      <td>0.012121</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>645</th>\n",
       "      <td>-0.036973</td>\n",
       "      <td>-0.068258</td>\n",
       "      <td>0.080347</td>\n",
       "      <td>0.056737</td>\n",
       "      <td>-0.121469</td>\n",
       "      <td>0.012606</td>\n",
       "      <td>0.072143</td>\n",
       "      <td>-0.181021</td>\n",
       "      <td>-0.051429</td>\n",
       "      <td>0.074363</td>\n",
       "      <td>...</td>\n",
       "      <td>0.234961</td>\n",
       "      <td>0.025535</td>\n",
       "      <td>-0.005260</td>\n",
       "      <td>0.291611</td>\n",
       "      <td>0.384691</td>\n",
       "      <td>-0.007476</td>\n",
       "      <td>-0.097842</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.010586</td>\n",
       "      <td>-0.131849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>646</th>\n",
       "      <td>0.028265</td>\n",
       "      <td>0.202899</td>\n",
       "      <td>0.111763</td>\n",
       "      <td>-0.233748</td>\n",
       "      <td>0.038715</td>\n",
       "      <td>0.088365</td>\n",
       "      <td>0.226196</td>\n",
       "      <td>-0.125008</td>\n",
       "      <td>0.300624</td>\n",
       "      <td>-0.298867</td>\n",
       "      <td>...</td>\n",
       "      <td>0.369234</td>\n",
       "      <td>0.205502</td>\n",
       "      <td>0.174752</td>\n",
       "      <td>0.181013</td>\n",
       "      <td>-0.252004</td>\n",
       "      <td>0.207629</td>\n",
       "      <td>0.183584</td>\n",
       "      <td>0.010586</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.319675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>647</th>\n",
       "      <td>-0.382432</td>\n",
       "      <td>0.455165</td>\n",
       "      <td>0.141706</td>\n",
       "      <td>-0.494055</td>\n",
       "      <td>-0.243578</td>\n",
       "      <td>0.072026</td>\n",
       "      <td>0.706769</td>\n",
       "      <td>-0.361463</td>\n",
       "      <td>0.662341</td>\n",
       "      <td>-0.817035</td>\n",
       "      <td>...</td>\n",
       "      <td>0.646631</td>\n",
       "      <td>0.590326</td>\n",
       "      <td>0.199716</td>\n",
       "      <td>0.204419</td>\n",
       "      <td>-0.574426</td>\n",
       "      <td>0.803752</td>\n",
       "      <td>0.012121</td>\n",
       "      <td>-0.131849</td>\n",
       "      <td>0.319675</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>648 rows × 648 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           60        74        85        95       114       169       199  \\\n",
       "0    1.000000 -0.203803 -0.041840  0.135602  0.297998 -0.025113 -0.373148   \n",
       "1   -0.203803  1.000000  0.084412 -0.320546 -0.071207  0.101266  0.424646   \n",
       "2   -0.041840  0.084412  1.000000 -0.045254 -0.170664 -0.124421  0.323632   \n",
       "3    0.135602 -0.320546 -0.045254  1.000000  0.097013  0.056162 -0.387976   \n",
       "4    0.297998 -0.071207 -0.170664  0.097013  1.000000  0.076526 -0.309533   \n",
       "..        ...       ...       ...       ...       ...       ...       ...   \n",
       "643 -0.417258  0.370390  0.115981 -0.399549 -0.314652  0.030274  0.669776   \n",
       "644  0.143673  0.104439  0.112563 -0.029409  0.182568  0.087487  0.010110   \n",
       "645 -0.036973 -0.068258  0.080347  0.056737 -0.121469  0.012606  0.072143   \n",
       "646  0.028265  0.202899  0.111763 -0.233748  0.038715  0.088365  0.226196   \n",
       "647 -0.382432  0.455165  0.141706 -0.494055 -0.243578  0.072026  0.706769   \n",
       "\n",
       "          415       562       657  ...     43720     43830     43873  \\\n",
       "0    0.335764 -0.295790  0.397197  ... -0.309133 -0.326422 -0.096837   \n",
       "1   -0.175569  0.446181 -0.486499  ...  0.386614  0.424727  0.153691   \n",
       "2   -0.062687  0.161443 -0.193569  ...  0.360204  0.271762  0.036250   \n",
       "3    0.232699 -0.433461  0.467508  ... -0.401847 -0.356511 -0.150706   \n",
       "4    0.360927 -0.220001  0.279083  ... -0.307539 -0.251303  0.034189   \n",
       "..        ...       ...       ...  ...       ...       ...       ...   \n",
       "643 -0.405258  0.643508 -0.785690  ...  0.605081  0.549997  0.256985   \n",
       "644  0.133939  0.118709 -0.055325  ...  0.083770 -0.110514  0.022980   \n",
       "645 -0.181021 -0.051429  0.074363  ...  0.234961  0.025535 -0.005260   \n",
       "646 -0.125008  0.300624 -0.298867  ...  0.369234  0.205502  0.174752   \n",
       "647 -0.361463  0.662341 -0.817035  ...  0.646631  0.590326  0.199716   \n",
       "\n",
       "        44001     44033     44111     44119     44155     44221     44254  \n",
       "0   -0.186167  0.136122 -0.417258  0.143673 -0.036973  0.028265 -0.382432  \n",
       "1    0.069929 -0.348278  0.370390  0.104439 -0.068258  0.202899  0.455165  \n",
       "2   -0.038425  0.057585  0.115981  0.112563  0.080347  0.111763  0.141706  \n",
       "3   -0.066169  0.363566 -0.399549 -0.029409  0.056737 -0.233748 -0.494055  \n",
       "4   -0.165472 -0.075354 -0.314652  0.182568 -0.121469  0.038715 -0.243578  \n",
       "..        ...       ...       ...       ...       ...       ...       ...  \n",
       "643  0.361084 -0.449045  1.000000 -0.084839 -0.007476  0.207629  0.803752  \n",
       "644 -0.062397 -0.051659 -0.084839  1.000000 -0.097842  0.183584  0.012121  \n",
       "645  0.291611  0.384691 -0.007476 -0.097842  1.000000  0.010586 -0.131849  \n",
       "646  0.181013 -0.252004  0.207629  0.183584  0.010586  1.000000  0.319675  \n",
       "647  0.204419 -0.574426  0.803752  0.012121 -0.131849  0.319675  1.000000  \n",
       "\n",
       "[648 rows x 648 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cutoff = 10000\n",
    "corr_meth = 'pearson'\n",
    "corr_csv = '{}_gene_correlation_{}.csv'.format(corr_meth,cutoff)\n",
    "\n",
    "corr = pd.read_csv(src + corr_csv)\n",
    "corr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Retrieve the metadata corresponding to the highly varying genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 (648, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>position_left</th>\n",
       "      <th>position_right</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>239383320</td>\n",
       "      <td>239384004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>5</td>\n",
       "      <td>gene</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>Zm00001d036567</td>\n",
       "      <td>6</td>\n",
       "      <td>gene</td>\n",
       "      <td>92808874</td>\n",
       "      <td>92811320</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>Zm00001d013039</td>\n",
       "      <td>5</td>\n",
       "      <td>gene</td>\n",
       "      <td>3603516</td>\n",
       "      <td>3604227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>Zm00001d051174</td>\n",
       "      <td>4</td>\n",
       "      <td>gene</td>\n",
       "      <td>147173977</td>\n",
       "      <td>147176860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44111</th>\n",
       "      <td>Zm00001d021421</td>\n",
       "      <td>7</td>\n",
       "      <td>gene</td>\n",
       "      <td>151705668</td>\n",
       "      <td>151712216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44119</th>\n",
       "      <td>Zm00001d053593</td>\n",
       "      <td>4</td>\n",
       "      <td>gene</td>\n",
       "      <td>235759773</td>\n",
       "      <td>235760235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44155</th>\n",
       "      <td>Zm00001d027442</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>5434307</td>\n",
       "      <td>5435871</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44221</th>\n",
       "      <td>Zm00001d030725</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>155959517</td>\n",
       "      <td>155963384</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44254</th>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>648 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 gene  chromosome feature_type  position_left  position_right\n",
       "60     Zm00001d032825           1         gene      239383320       239384004\n",
       "74     Zm00001d014957           5         gene       69975364        69978349\n",
       "85     Zm00001d036567           6         gene       92808874        92811320\n",
       "95     Zm00001d013039           5         gene        3603516         3604227\n",
       "114    Zm00001d051174           4         gene      147173977       147176860\n",
       "...               ...         ...          ...            ...             ...\n",
       "44111  Zm00001d021421           7         gene      151705668       151712216\n",
       "44119  Zm00001d053593           4         gene      235759773       235760235\n",
       "44155  Zm00001d027442           1         gene        5434307         5435871\n",
       "44221  Zm00001d030725           1         gene      155959517       155963384\n",
       "44254  Zm00001d028653           1         gene       41833737        41838204\n",
       "\n",
       "[648 rows x 5 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = gene_info.iloc[corr.columns]\n",
    "foo = meta.chromosome.unique()\n",
    "print(len(foo), meta.shape)\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d047861</td>\n",
       "      <td>-15.244080</td>\n",
       "      <td>4.703940e-47</td>\n",
       "      <td>8.784256e-43</td>\n",
       "      <td>-0.012608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d004808</td>\n",
       "      <td>-10.441780</td>\n",
       "      <td>3.181453e-24</td>\n",
       "      <td>2.339655e-20</td>\n",
       "      <td>-3.733139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d031538</td>\n",
       "      <td>-10.165090</td>\n",
       "      <td>4.220149e-23</td>\n",
       "      <td>2.880657e-19</td>\n",
       "      <td>-12.332500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d008253</td>\n",
       "      <td>-10.165090</td>\n",
       "      <td>4.220149e-23</td>\n",
       "      <td>2.880657e-19</td>\n",
       "      <td>-13.845900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d032862</td>\n",
       "      <td>-10.165090</td>\n",
       "      <td>4.220149e-23</td>\n",
       "      <td>2.880657e-19</td>\n",
       "      <td>-0.000578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196037</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d001752</td>\n",
       "      <td>-7.556017</td>\n",
       "      <td>9.846873e-14</td>\n",
       "      <td>3.089405e-10</td>\n",
       "      <td>-0.006062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196038</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d001652</td>\n",
       "      <td>-7.371548</td>\n",
       "      <td>3.694633e-13</td>\n",
       "      <td>1.089526e-09</td>\n",
       "      <td>-0.042778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196039</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d045638</td>\n",
       "      <td>-7.322923</td>\n",
       "      <td>5.210988e-13</td>\n",
       "      <td>1.501189e-09</td>\n",
       "      <td>-6.382181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196040</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d031869</td>\n",
       "      <td>-7.303307</td>\n",
       "      <td>5.983143e-13</td>\n",
       "      <td>1.712528e-09</td>\n",
       "      <td>-0.196164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196041</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d026009</td>\n",
       "      <td>-7.003921</td>\n",
       "      <td>4.738112e-12</td>\n",
       "      <td>1.214845e-08</td>\n",
       "      <td>-0.015592</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17196042 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "0              chr1         44288        49837  Zm00001d027230         gene   \n",
       "1              chr1         44288        49837  Zm00001d027230         gene   \n",
       "2              chr1         44288        49837  Zm00001d027230         gene   \n",
       "3              chr1         44288        49837  Zm00001d027230         gene   \n",
       "4              chr1         44288        49837  Zm00001d027230         gene   \n",
       "...             ...           ...          ...             ...          ...   \n",
       "17196037       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196038       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196039       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196040       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196041       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "\n",
       "         SNP_Chr  SNP_Start   SNP_Stop         SNP_ID Associated_Gene_ID  \\\n",
       "0           chr1      44306      44307      rs1_44306     Zm00001d047861   \n",
       "1           chr1      44306      44307      rs1_44306     Zm00001d004808   \n",
       "2           chr1      44306      44307      rs1_44306     Zm00001d031538   \n",
       "3           chr1      44306      44307      rs1_44306     Zm00001d008253   \n",
       "4           chr1      44306      44307      rs1_44306     Zm00001d032862   \n",
       "...          ...        ...        ...            ...                ...   \n",
       "17196037    chr9  159690349  159690350  rs9_159690349     Zm00001d001752   \n",
       "17196038    chr9  159690349  159690350  rs9_159690349     Zm00001d001652   \n",
       "17196039    chr9  159690349  159690350  rs9_159690349     Zm00001d045638   \n",
       "17196040    chr9  159690349  159690350  rs9_159690349     Zm00001d031869   \n",
       "17196041    chr9  159690349  159690350  rs9_159690349     Zm00001d026009   \n",
       "\n",
       "          Statistic          PVal           FDR       Beta  \n",
       "0        -15.244080  4.703940e-47  8.784256e-43  -0.012608  \n",
       "1        -10.441780  3.181453e-24  2.339655e-20  -3.733139  \n",
       "2        -10.165090  4.220149e-23  2.880657e-19 -12.332500  \n",
       "3        -10.165090  4.220149e-23  2.880657e-19 -13.845900  \n",
       "4        -10.165090  4.220149e-23  2.880657e-19  -0.000578  \n",
       "...             ...           ...           ...        ...  \n",
       "17196037  -7.556017  9.846873e-14  3.089405e-10  -0.006062  \n",
       "17196038  -7.371548  3.694633e-13  1.089526e-09  -0.042778  \n",
       "17196039  -7.322923  5.210988e-13  1.501189e-09  -6.382181  \n",
       "17196040  -7.303307  5.983143e-13  1.712528e-09  -0.196164  \n",
       "17196041  -7.003921  4.738112e-12  1.214845e-08  -0.015592  \n",
       "\n",
       "[17196042 rows x 14 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl = pd.read_table(eqtl_file)\n",
    "eqtl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chr1', 'chr10', 'chr2', 'chr3', 'chr4', 'chr5', 'chr6', 'chr7',\n",
       "       'chr8', 'chr9'], dtype=object)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl.R_Gene_Chr.unique()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Making subsets of the eQTL data\n",
    "\n",
    "We subsample the eQTL matrix above. \n",
    "+ Consider only the eQTLs where the SNP is located in one of the highly varying genes.\n",
    "+ Consider only the eQTLs where the SNP is associated to one of the highly varying genes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1907240</th>\n",
       "      <td>chr1</td>\n",
       "      <td>239383320</td>\n",
       "      <td>239384004</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>239383364</td>\n",
       "      <td>239383365</td>\n",
       "      <td>rs1_239383364</td>\n",
       "      <td>Zm00001d015191</td>\n",
       "      <td>-13.667570</td>\n",
       "      <td>6.379975e-39</td>\n",
       "      <td>9.103020e-35</td>\n",
       "      <td>-0.027237</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1907241</th>\n",
       "      <td>chr1</td>\n",
       "      <td>239383320</td>\n",
       "      <td>239384004</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>239383364</td>\n",
       "      <td>239383365</td>\n",
       "      <td>rs1_239383364</td>\n",
       "      <td>Zm00001d032949</td>\n",
       "      <td>-13.667570</td>\n",
       "      <td>6.379975e-39</td>\n",
       "      <td>9.103020e-35</td>\n",
       "      <td>-0.016791</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1907242</th>\n",
       "      <td>chr1</td>\n",
       "      <td>239383320</td>\n",
       "      <td>239384004</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>239383364</td>\n",
       "      <td>239383365</td>\n",
       "      <td>rs1_239383364</td>\n",
       "      <td>Zm00001d044935</td>\n",
       "      <td>-13.667570</td>\n",
       "      <td>6.379975e-39</td>\n",
       "      <td>9.103020e-35</td>\n",
       "      <td>-0.001824</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1907243</th>\n",
       "      <td>chr1</td>\n",
       "      <td>239383320</td>\n",
       "      <td>239384004</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>239383364</td>\n",
       "      <td>239383365</td>\n",
       "      <td>rs1_239383364</td>\n",
       "      <td>Zm00001d019017</td>\n",
       "      <td>-13.667570</td>\n",
       "      <td>6.379975e-39</td>\n",
       "      <td>9.103020e-35</td>\n",
       "      <td>-0.011985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1907244</th>\n",
       "      <td>chr1</td>\n",
       "      <td>239383320</td>\n",
       "      <td>239384004</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>239383364</td>\n",
       "      <td>239383365</td>\n",
       "      <td>rs1_239383364</td>\n",
       "      <td>Zm00001d042657</td>\n",
       "      <td>-13.667570</td>\n",
       "      <td>6.379975e-39</td>\n",
       "      <td>9.103020e-35</td>\n",
       "      <td>-0.020081</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617571</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837857</td>\n",
       "      <td>41837858</td>\n",
       "      <td>rs1_41837857</td>\n",
       "      <td>Zm00001d019489</td>\n",
       "      <td>-7.031328</td>\n",
       "      <td>3.932666e-12</td>\n",
       "      <td>1.017631e-08</td>\n",
       "      <td>-0.006891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617572</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837857</td>\n",
       "      <td>41837858</td>\n",
       "      <td>rs1_41837857</td>\n",
       "      <td>Zm00001d001706</td>\n",
       "      <td>-6.838447</td>\n",
       "      <td>1.439989e-11</td>\n",
       "      <td>3.450679e-08</td>\n",
       "      <td>-0.048347</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617573</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837857</td>\n",
       "      <td>41837858</td>\n",
       "      <td>rs1_41837857</td>\n",
       "      <td>Zm00001d030802</td>\n",
       "      <td>-6.737078</td>\n",
       "      <td>2.812868e-11</td>\n",
       "      <td>6.455689e-08</td>\n",
       "      <td>-0.129585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617574</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837857</td>\n",
       "      <td>41837858</td>\n",
       "      <td>rs1_41837857</td>\n",
       "      <td>Zm00001d033812</td>\n",
       "      <td>-6.726736</td>\n",
       "      <td>3.010252e-11</td>\n",
       "      <td>6.881629e-08</td>\n",
       "      <td>-0.006297</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617575</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837857</td>\n",
       "      <td>41837858</td>\n",
       "      <td>rs1_41837857</td>\n",
       "      <td>Zm00001d039242</td>\n",
       "      <td>-6.686576</td>\n",
       "      <td>3.913931e-11</td>\n",
       "      <td>8.810140e-08</td>\n",
       "      <td>-0.085344</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>570464 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "1907240       chr1     239383320    239384004  Zm00001d032825         gene   \n",
       "1907241       chr1     239383320    239384004  Zm00001d032825         gene   \n",
       "1907242       chr1     239383320    239384004  Zm00001d032825         gene   \n",
       "1907243       chr1     239383320    239384004  Zm00001d032825         gene   \n",
       "1907244       chr1     239383320    239384004  Zm00001d032825         gene   \n",
       "...            ...           ...          ...             ...          ...   \n",
       "617571        chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617572        chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617573        chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617574        chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617575        chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "\n",
       "        SNP_Chr  SNP_Start   SNP_Stop         SNP_ID Associated_Gene_ID  \\\n",
       "1907240    chr1  239383364  239383365  rs1_239383364     Zm00001d015191   \n",
       "1907241    chr1  239383364  239383365  rs1_239383364     Zm00001d032949   \n",
       "1907242    chr1  239383364  239383365  rs1_239383364     Zm00001d044935   \n",
       "1907243    chr1  239383364  239383365  rs1_239383364     Zm00001d019017   \n",
       "1907244    chr1  239383364  239383365  rs1_239383364     Zm00001d042657   \n",
       "...         ...        ...        ...            ...                ...   \n",
       "617571     chr1   41837857   41837858   rs1_41837857     Zm00001d019489   \n",
       "617572     chr1   41837857   41837858   rs1_41837857     Zm00001d001706   \n",
       "617573     chr1   41837857   41837858   rs1_41837857     Zm00001d030802   \n",
       "617574     chr1   41837857   41837858   rs1_41837857     Zm00001d033812   \n",
       "617575     chr1   41837857   41837858   rs1_41837857     Zm00001d039242   \n",
       "\n",
       "         Statistic          PVal           FDR      Beta  \n",
       "1907240 -13.667570  6.379975e-39  9.103020e-35 -0.027237  \n",
       "1907241 -13.667570  6.379975e-39  9.103020e-35 -0.016791  \n",
       "1907242 -13.667570  6.379975e-39  9.103020e-35 -0.001824  \n",
       "1907243 -13.667570  6.379975e-39  9.103020e-35 -0.011985  \n",
       "1907244 -13.667570  6.379975e-39  9.103020e-35 -0.020081  \n",
       "...            ...           ...           ...       ...  \n",
       "617571   -7.031328  3.932666e-12  1.017631e-08 -0.006891  \n",
       "617572   -6.838447  1.439989e-11  3.450679e-08 -0.048347  \n",
       "617573   -6.737078  2.812868e-11  6.455689e-08 -0.129585  \n",
       "617574   -6.726736  3.010252e-11  6.881629e-08 -0.006297  \n",
       "617575   -6.686576  3.913931e-11  8.810140e-08 -0.085344  \n",
       "\n",
       "[570464 rows x 14 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "i = 0\n",
    "located = pd.DataFrame()\n",
    "\n",
    "for i in range(meta.shape[0]):\n",
    "    located = located.append(eqtl[ eqtl['R_Gene_ID'] == meta.iloc[i]['gene']] , sort=False)\n",
    "\n",
    "located"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "639 genes out of 648 contain associated SNPs\n"
     ]
    }
   ],
   "source": [
    "print(len(located.R_Gene_ID.unique()), 'genes out of', len(meta.gene.unique()),\n",
    "      'contain associated SNPs')\n",
    "located.drop(columns=['SNP_Chr','SNP_Stop','R_Gene_Class']).to_csv(dst+'eqtl_located_{}.csv'.format(cutoff),\n",
    "                                                                  index = True, index_label='Original_Index')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46166</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3095307</td>\n",
       "      <td>3095604</td>\n",
       "      <td>Zm00001d027332</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3095600</td>\n",
       "      <td>3095601</td>\n",
       "      <td>rs1_3095600</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89558</th>\n",
       "      <td>chr1</td>\n",
       "      <td>4494974</td>\n",
       "      <td>4499451</td>\n",
       "      <td>Zm00001d027403</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4495369</td>\n",
       "      <td>4495370</td>\n",
       "      <td>rs1_4495369</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95346</th>\n",
       "      <td>chr1</td>\n",
       "      <td>4891940</td>\n",
       "      <td>4894990</td>\n",
       "      <td>Zm00001d027419</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4892252</td>\n",
       "      <td>4892253</td>\n",
       "      <td>rs1_4892252</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122615</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5819751</td>\n",
       "      <td>5825029</td>\n",
       "      <td>Zm00001d027462</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5824650</td>\n",
       "      <td>5824651</td>\n",
       "      <td>rs1_5824650</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142221</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6407152</td>\n",
       "      <td>6409815</td>\n",
       "      <td>Zm00001d027488</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6409375</td>\n",
       "      <td>6409376</td>\n",
       "      <td>rs1_6409375</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15900382</th>\n",
       "      <td>chr8</td>\n",
       "      <td>178335675</td>\n",
       "      <td>178339115</td>\n",
       "      <td>Zm00001d012670</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr8</td>\n",
       "      <td>178336921</td>\n",
       "      <td>178336922</td>\n",
       "      <td>rs8_178336921</td>\n",
       "      <td>Zm00001d013766</td>\n",
       "      <td>-9.052595</td>\n",
       "      <td>7.876414e-19</td>\n",
       "      <td>4.003547e-15</td>\n",
       "      <td>-816.8673</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16337373</th>\n",
       "      <td>chr9</td>\n",
       "      <td>57489278</td>\n",
       "      <td>57491429</td>\n",
       "      <td>Zm00001d046036</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>57489366</td>\n",
       "      <td>57489367</td>\n",
       "      <td>rs9_57489366</td>\n",
       "      <td>Zm00001d013766</td>\n",
       "      <td>-7.915426</td>\n",
       "      <td>6.914868e-15</td>\n",
       "      <td>2.463223e-11</td>\n",
       "      <td>-456.7894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17099343</th>\n",
       "      <td>chr9</td>\n",
       "      <td>155994046</td>\n",
       "      <td>155999070</td>\n",
       "      <td>Zm00001d048424</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>155996738</td>\n",
       "      <td>155996739</td>\n",
       "      <td>rs9_155996738</td>\n",
       "      <td>Zm00001d013766</td>\n",
       "      <td>-7.062233</td>\n",
       "      <td>3.185022e-12</td>\n",
       "      <td>8.335277e-09</td>\n",
       "      <td>-915.2276</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17160578</th>\n",
       "      <td>chr9</td>\n",
       "      <td>157608353</td>\n",
       "      <td>157609505</td>\n",
       "      <td>Zm00001d048513</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>157609219</td>\n",
       "      <td>157609220</td>\n",
       "      <td>rs9_157609219</td>\n",
       "      <td>Zm00001d013766</td>\n",
       "      <td>-6.979505</td>\n",
       "      <td>5.590658e-12</td>\n",
       "      <td>1.419101e-08</td>\n",
       "      <td>-905.0377</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3291307</th>\n",
       "      <td>chr10</td>\n",
       "      <td>105692425</td>\n",
       "      <td>105720443</td>\n",
       "      <td>Zm00001d025130</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr10</td>\n",
       "      <td>105720343</td>\n",
       "      <td>105720344</td>\n",
       "      <td>rs10_105720343</td>\n",
       "      <td>Zm00001d049650</td>\n",
       "      <td>-6.720462</td>\n",
       "      <td>3.136558e-11</td>\n",
       "      <td>7.152035e-08</td>\n",
       "      <td>-1772.8550</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>30147 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "46166          chr1       3095307      3095604  Zm00001d027332         gene   \n",
       "89558          chr1       4494974      4499451  Zm00001d027403         gene   \n",
       "95346          chr1       4891940      4894990  Zm00001d027419         gene   \n",
       "122615         chr1       5819751      5825029  Zm00001d027462         gene   \n",
       "142221         chr1       6407152      6409815  Zm00001d027488         gene   \n",
       "...             ...           ...          ...             ...          ...   \n",
       "15900382       chr8     178335675    178339115  Zm00001d012670         gene   \n",
       "16337373       chr9      57489278     57491429  Zm00001d046036         gene   \n",
       "17099343       chr9     155994046    155999070  Zm00001d048424         gene   \n",
       "17160578       chr9     157608353    157609505  Zm00001d048513         gene   \n",
       "3291307       chr10     105692425    105720443  Zm00001d025130         gene   \n",
       "\n",
       "         SNP_Chr  SNP_Start   SNP_Stop          SNP_ID Associated_Gene_ID  \\\n",
       "46166       chr1    3095600    3095601     rs1_3095600     Zm00001d032825   \n",
       "89558       chr1    4495369    4495370     rs1_4495369     Zm00001d032825   \n",
       "95346       chr1    4892252    4892253     rs1_4892252     Zm00001d032825   \n",
       "122615      chr1    5824650    5824651     rs1_5824650     Zm00001d032825   \n",
       "142221      chr1    6409375    6409376     rs1_6409375     Zm00001d032825   \n",
       "...          ...        ...        ...             ...                ...   \n",
       "15900382    chr8  178336921  178336922   rs8_178336921     Zm00001d013766   \n",
       "16337373    chr9   57489366   57489367    rs9_57489366     Zm00001d013766   \n",
       "17099343    chr9  155996738  155996739   rs9_155996738     Zm00001d013766   \n",
       "17160578    chr9  157609219  157609220   rs9_157609219     Zm00001d013766   \n",
       "3291307    chr10  105720343  105720344  rs10_105720343     Zm00001d049650   \n",
       "\n",
       "          Statistic          PVal           FDR       Beta  \n",
       "46166     -6.583017  7.653515e-11  1.648439e-07  -618.7353  \n",
       "89558     -6.583017  7.653515e-11  1.648439e-07  -618.7353  \n",
       "95346     -6.583017  7.653515e-11  1.648439e-07  -618.7353  \n",
       "122615    -6.583017  7.653515e-11  1.648439e-07  -618.7353  \n",
       "142221    -6.583017  7.653515e-11  1.648439e-07  -618.7353  \n",
       "...             ...           ...           ...        ...  \n",
       "15900382  -9.052595  7.876414e-19  4.003547e-15  -816.8673  \n",
       "16337373  -7.915426  6.914868e-15  2.463223e-11  -456.7894  \n",
       "17099343  -7.062233  3.185022e-12  8.335277e-09  -915.2276  \n",
       "17160578  -6.979505  5.590658e-12  1.419101e-08  -905.0377  \n",
       "3291307   -6.720462  3.136558e-11  7.152035e-08 -1772.8550  \n",
       "\n",
       "[30147 rows x 14 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "related = pd.DataFrame()\n",
    "\n",
    "for i in range(meta.shape[0]):\n",
    "    related = related.append(eqtl[ eqtl['Associated_Gene_ID'] == meta.iloc[i]['gene']], sort=False)\n",
    "\n",
    "related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "133 genes out of 648 contain associated SNPs\n"
     ]
    }
   ],
   "source": [
    "print(len(related.Associated_Gene_ID.unique()), 'genes out of', len(meta.gene.unique()),\n",
    "      'contain associated SNPs')\n",
    "related.drop(columns=['SNP_Chr','SNP_Stop','R_Gene_Class']).to_csv(dst+'eqtl_related_{}.csv'.format(cutoff),\n",
    "                                                                  index = True, index_label='Original_Index')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10417261</th>\n",
       "      <td>chr5</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr5</td>\n",
       "      <td>69975582</td>\n",
       "      <td>69975583</td>\n",
       "      <td>rs5_69975582</td>\n",
       "      <td>Zm00001d040202</td>\n",
       "      <td>-7.155506</td>\n",
       "      <td>1.677400e-12</td>\n",
       "      <td>4.564204e-09</td>\n",
       "      <td>-580.0772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417335</th>\n",
       "      <td>chr5</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr5</td>\n",
       "      <td>69975588</td>\n",
       "      <td>69975589</td>\n",
       "      <td>rs5_69975588</td>\n",
       "      <td>Zm00001d040202</td>\n",
       "      <td>-7.563082</td>\n",
       "      <td>9.355389e-14</td>\n",
       "      <td>2.941734e-10</td>\n",
       "      <td>-611.2713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417910</th>\n",
       "      <td>chr5</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr5</td>\n",
       "      <td>69978178</td>\n",
       "      <td>69978179</td>\n",
       "      <td>rs5_69978178</td>\n",
       "      <td>Zm00001d040202</td>\n",
       "      <td>-7.563082</td>\n",
       "      <td>9.355389e-14</td>\n",
       "      <td>2.941734e-10</td>\n",
       "      <td>-611.2713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12014337</th>\n",
       "      <td>chr6</td>\n",
       "      <td>92808874</td>\n",
       "      <td>92811320</td>\n",
       "      <td>Zm00001d036567</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr6</td>\n",
       "      <td>92811304</td>\n",
       "      <td>92811305</td>\n",
       "      <td>rs6_92811304</td>\n",
       "      <td>Zm00001d001139</td>\n",
       "      <td>-8.917276</td>\n",
       "      <td>2.446043e-18</td>\n",
       "      <td>1.199951e-14</td>\n",
       "      <td>-1262.5240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14255497</th>\n",
       "      <td>chr7</td>\n",
       "      <td>173953536</td>\n",
       "      <td>173954457</td>\n",
       "      <td>Zm00001d022264</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr7</td>\n",
       "      <td>173954253</td>\n",
       "      <td>173954254</td>\n",
       "      <td>rs7_173954253</td>\n",
       "      <td>Zm00001d044495</td>\n",
       "      <td>-8.451948</td>\n",
       "      <td>1.081581e-16</td>\n",
       "      <td>4.604894e-13</td>\n",
       "      <td>-823.7607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616952</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41836775</td>\n",
       "      <td>41836776</td>\n",
       "      <td>rs1_41836775</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.789329</td>\n",
       "      <td>7.050030e-18</td>\n",
       "      <td>3.306002e-14</td>\n",
       "      <td>-288.7677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616999</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41836824</td>\n",
       "      <td>41836825</td>\n",
       "      <td>rs1_41836824</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.789329</td>\n",
       "      <td>7.050030e-18</td>\n",
       "      <td>3.306002e-14</td>\n",
       "      <td>-288.7677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617046</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41836963</td>\n",
       "      <td>41836964</td>\n",
       "      <td>rs1_41836963</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.789329</td>\n",
       "      <td>7.050030e-18</td>\n",
       "      <td>3.306002e-14</td>\n",
       "      <td>-288.7677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617075</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837097</td>\n",
       "      <td>41837098</td>\n",
       "      <td>rs1_41837097</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.439968</td>\n",
       "      <td>1.189764e-16</td>\n",
       "      <td>5.001608e-13</td>\n",
       "      <td>-268.1414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617171</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837500</td>\n",
       "      <td>41837501</td>\n",
       "      <td>rs1_41837500</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-9.638586</td>\n",
       "      <td>4.965583e-21</td>\n",
       "      <td>2.961445e-17</td>\n",
       "      <td>-341.2709</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1007 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "10417261       chr5      69975364     69978349  Zm00001d014957         gene   \n",
       "10417335       chr5      69975364     69978349  Zm00001d014957         gene   \n",
       "10417910       chr5      69975364     69978349  Zm00001d014957         gene   \n",
       "12014337       chr6      92808874     92811320  Zm00001d036567         gene   \n",
       "14255497       chr7     173953536    173954457  Zm00001d022264         gene   \n",
       "...             ...           ...          ...             ...          ...   \n",
       "616952         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "616999         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617046         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617075         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617171         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "\n",
       "         SNP_Chr  SNP_Start   SNP_Stop         SNP_ID Associated_Gene_ID  \\\n",
       "10417261    chr5   69975582   69975583   rs5_69975582     Zm00001d040202   \n",
       "10417335    chr5   69975588   69975589   rs5_69975588     Zm00001d040202   \n",
       "10417910    chr5   69978178   69978179   rs5_69978178     Zm00001d040202   \n",
       "12014337    chr6   92811304   92811305   rs6_92811304     Zm00001d001139   \n",
       "14255497    chr7  173954253  173954254  rs7_173954253     Zm00001d044495   \n",
       "...          ...        ...        ...            ...                ...   \n",
       "616952      chr1   41836775   41836776   rs1_41836775         zma-MIR482   \n",
       "616999      chr1   41836824   41836825   rs1_41836824         zma-MIR482   \n",
       "617046      chr1   41836963   41836964   rs1_41836963         zma-MIR482   \n",
       "617075      chr1   41837097   41837098   rs1_41837097         zma-MIR482   \n",
       "617171      chr1   41837500   41837501   rs1_41837500         zma-MIR482   \n",
       "\n",
       "          Statistic          PVal           FDR       Beta  \n",
       "10417261  -7.155506  1.677400e-12  4.564204e-09  -580.0772  \n",
       "10417335  -7.563082  9.355389e-14  2.941734e-10  -611.2713  \n",
       "10417910  -7.563082  9.355389e-14  2.941734e-10  -611.2713  \n",
       "12014337  -8.917276  2.446043e-18  1.199951e-14 -1262.5240  \n",
       "14255497  -8.451948  1.081581e-16  4.604894e-13  -823.7607  \n",
       "...             ...           ...           ...        ...  \n",
       "616952    -8.789329  7.050030e-18  3.306002e-14  -288.7677  \n",
       "616999    -8.789329  7.050030e-18  3.306002e-14  -288.7677  \n",
       "617046    -8.789329  7.050030e-18  3.306002e-14  -288.7677  \n",
       "617075    -8.439968  1.189764e-16  5.001608e-13  -268.1414  \n",
       "617171    -9.638586  4.965583e-21  2.961445e-17  -341.2709  \n",
       "\n",
       "[1007 rows x 14 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "locrel = pd.concat([located, related], axis=1, join='inner')\n",
    "locrel = locrel.iloc[:, :located.shape[1]]\n",
    "locrel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92 genes out of 648 contain associated SNPs\n"
     ]
    }
   ],
   "source": [
    "print(len(locrel.Associated_Gene_ID.unique()), 'genes out of', len(meta.gene.unique()),\n",
    "      'contain associated SNPs')\n",
    "related.drop(columns=['SNP_Chr','SNP_Stop','R_Gene_Class']).to_csv(dst+'eqtl_locrel_{}.csv'.format(cutoff),\n",
    "                                                                  index = True, index_label='Original_Index')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
