{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### <p style=\"text-align: right;\"> &#9989; **Context Matters** </p>\n",
    "#### <p style=\"text-align: right;\"> &#9989; Christina, Hao, Yunfei, Erik</p>\n",
    "\n",
    "# Module 3\n",
    "\n",
    "## How do gene transcription relate to phenotype?\n",
    "\n",
    "We will work with maize.\n",
    "\n",
    "### Preliminaries\n",
    "\n",
    "First, import the usual libraries\n",
    "- `math`: basic math operations\n",
    "- `os`: enable file manipulation with the OS\n",
    "- `sys`: enable interaction with commandline\n",
    "- `importlib` : reload libraries if necessary\n",
    "- `argparse` : pass arguments directly to the command line\n",
    "- `glob`: more variable manipulation\n",
    "- `matplotlib.pyplot`: default plotter (I personally like ggplot waaaaay better. E)\n",
    "    - `inline`: so that plots are shown in the notebook\n",
    "- `numpy`: all number cruching done here\n",
    "- `pandas`: data wrangling"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import math\n",
    "import importlib\n",
    "import numpy as np\n",
    "import os\n",
    "import sys\n",
    "import argparse\n",
    "import glob\n",
    "from matplotlib import pyplot as plt\n",
    "import matplotlib.cm as cm\n",
    "%matplotlib inline\n",
    "from mpl_toolkits.mplot3d import Axes3D\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "sns.set(color_codes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Specify the gene expression raw data file and a directory to save plots and CSVs.\n",
    "\n",
    "The eQTL file was provided by (Christina,Fabio,Nolan,Scott)-group in Module 1. Supposedly, it is the result of eQTL performed of every gene against every SNP."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# path to FPKM data\n",
    "gene_file = '~/documents/css893/942_FPKM_B73_genes_w_feature.txt'\n",
    "\n",
    "# path to eQTL data\n",
    "eqtl_file = '/home/ejam/documents/css893/Merged_eQTL_Data.tsv'\n",
    "\n",
    "# path to correlation matrix directory\n",
    "src = '/home/ejam/documents/css893/tpm_corr/'\n",
    "\n",
    "# path to save results\n",
    "dst = '/home/ejam/documents/css893/context_matters/results/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the original FPKM data. This time we only care of the metadata. All the contigs were treated as _chromosome 11_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ejam/.local/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (1) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  interactivity=interactivity, compiler=compiler, result=result)\n"
     ]
    }
   ],
   "source": [
    "fpkm = pd.read_table(gene_file)\n",
    "metadata_cols = 5\n",
    "\n",
    "gene_info = fpkm.iloc[:, :metadata_cols]\n",
    "foo = gene_info['chromosome'].unique()\n",
    "\n",
    "for idx in foo:\n",
    "    if idx == str(idx):\n",
    "        value = 0\n",
    "        if idx[0] == 'B':\n",
    "            value = int('73' + (idx.split('_')[-1])[3:])\n",
    "            value = 11\n",
    "        else:\n",
    "            value = int(idx)\n",
    "        \n",
    "        mask = gene_info.chromosome == idx\n",
    "        gene_info.loc[mask, 'chromosome'] = value"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Load the correlation matrix. The correlation method is _pearson_ and we only consider those genes whose TPM variance was above 10e5."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>19</th>\n",
       "      <th>20</th>\n",
       "      <th>27</th>\n",
       "      <th>30</th>\n",
       "      <th>33</th>\n",
       "      <th>38</th>\n",
       "      <th>40</th>\n",
       "      <th>49</th>\n",
       "      <th>...</th>\n",
       "      <th>44212</th>\n",
       "      <th>44218</th>\n",
       "      <th>44219</th>\n",
       "      <th>44221</th>\n",
       "      <th>44227</th>\n",
       "      <th>44228</th>\n",
       "      <th>44245</th>\n",
       "      <th>44250</th>\n",
       "      <th>44254</th>\n",
       "      <th>44256</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.603307</td>\n",
       "      <td>0.617355</td>\n",
       "      <td>0.067536</td>\n",
       "      <td>-0.152953</td>\n",
       "      <td>-0.580836</td>\n",
       "      <td>-0.638142</td>\n",
       "      <td>0.579787</td>\n",
       "      <td>0.688214</td>\n",
       "      <td>-0.597067</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.495718</td>\n",
       "      <td>0.561550</td>\n",
       "      <td>-0.420209</td>\n",
       "      <td>0.198508</td>\n",
       "      <td>0.301550</td>\n",
       "      <td>0.473162</td>\n",
       "      <td>-0.283920</td>\n",
       "      <td>-0.468367</td>\n",
       "      <td>0.686651</td>\n",
       "      <td>-0.191131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.603307</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.737379</td>\n",
       "      <td>0.083168</td>\n",
       "      <td>0.016714</td>\n",
       "      <td>-0.508527</td>\n",
       "      <td>-0.633081</td>\n",
       "      <td>0.682938</td>\n",
       "      <td>0.585571</td>\n",
       "      <td>-0.633340</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.499109</td>\n",
       "      <td>0.627078</td>\n",
       "      <td>-0.332796</td>\n",
       "      <td>0.378169</td>\n",
       "      <td>0.279490</td>\n",
       "      <td>0.657190</td>\n",
       "      <td>-0.368799</td>\n",
       "      <td>-0.403331</td>\n",
       "      <td>0.742548</td>\n",
       "      <td>-0.133403</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.617355</td>\n",
       "      <td>0.737379</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.149760</td>\n",
       "      <td>-0.043953</td>\n",
       "      <td>-0.636658</td>\n",
       "      <td>-0.691886</td>\n",
       "      <td>0.812817</td>\n",
       "      <td>0.686925</td>\n",
       "      <td>-0.688775</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.535990</td>\n",
       "      <td>0.765954</td>\n",
       "      <td>-0.308007</td>\n",
       "      <td>0.382220</td>\n",
       "      <td>0.363331</td>\n",
       "      <td>0.769484</td>\n",
       "      <td>-0.389583</td>\n",
       "      <td>-0.468688</td>\n",
       "      <td>0.803792</td>\n",
       "      <td>-0.062469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.067536</td>\n",
       "      <td>0.083168</td>\n",
       "      <td>0.149760</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.026165</td>\n",
       "      <td>-0.048954</td>\n",
       "      <td>-0.126808</td>\n",
       "      <td>0.224780</td>\n",
       "      <td>0.059721</td>\n",
       "      <td>-0.075750</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.085676</td>\n",
       "      <td>0.127682</td>\n",
       "      <td>0.022826</td>\n",
       "      <td>0.020554</td>\n",
       "      <td>0.131990</td>\n",
       "      <td>0.037556</td>\n",
       "      <td>-0.132367</td>\n",
       "      <td>0.026499</td>\n",
       "      <td>0.052534</td>\n",
       "      <td>0.197024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-0.152953</td>\n",
       "      <td>0.016714</td>\n",
       "      <td>-0.043953</td>\n",
       "      <td>-0.026165</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.191237</td>\n",
       "      <td>0.105725</td>\n",
       "      <td>-0.126072</td>\n",
       "      <td>-0.180447</td>\n",
       "      <td>0.100474</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.023838</td>\n",
       "      <td>-0.152780</td>\n",
       "      <td>0.287822</td>\n",
       "      <td>0.201723</td>\n",
       "      <td>-0.134249</td>\n",
       "      <td>0.075790</td>\n",
       "      <td>-0.063227</td>\n",
       "      <td>0.052689</td>\n",
       "      <td>-0.030109</td>\n",
       "      <td>0.114066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7354</th>\n",
       "      <td>0.473162</td>\n",
       "      <td>0.657190</td>\n",
       "      <td>0.769484</td>\n",
       "      <td>0.037556</td>\n",
       "      <td>0.075790</td>\n",
       "      <td>-0.516658</td>\n",
       "      <td>-0.582472</td>\n",
       "      <td>0.677102</td>\n",
       "      <td>0.580988</td>\n",
       "      <td>-0.595140</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.461170</td>\n",
       "      <td>0.664644</td>\n",
       "      <td>-0.245511</td>\n",
       "      <td>0.377184</td>\n",
       "      <td>0.226978</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.322122</td>\n",
       "      <td>-0.396342</td>\n",
       "      <td>0.678366</td>\n",
       "      <td>-0.111052</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7355</th>\n",
       "      <td>-0.283920</td>\n",
       "      <td>-0.368799</td>\n",
       "      <td>-0.389583</td>\n",
       "      <td>-0.132367</td>\n",
       "      <td>-0.063227</td>\n",
       "      <td>0.286439</td>\n",
       "      <td>0.371946</td>\n",
       "      <td>-0.387424</td>\n",
       "      <td>-0.297418</td>\n",
       "      <td>0.385382</td>\n",
       "      <td>...</td>\n",
       "      <td>0.310319</td>\n",
       "      <td>-0.332960</td>\n",
       "      <td>0.091233</td>\n",
       "      <td>-0.218509</td>\n",
       "      <td>-0.210024</td>\n",
       "      <td>-0.322122</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.280048</td>\n",
       "      <td>-0.368306</td>\n",
       "      <td>-0.032168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7356</th>\n",
       "      <td>-0.468367</td>\n",
       "      <td>-0.403331</td>\n",
       "      <td>-0.468688</td>\n",
       "      <td>0.026499</td>\n",
       "      <td>0.052689</td>\n",
       "      <td>0.543811</td>\n",
       "      <td>0.491412</td>\n",
       "      <td>-0.438147</td>\n",
       "      <td>-0.429403</td>\n",
       "      <td>0.433600</td>\n",
       "      <td>...</td>\n",
       "      <td>0.429828</td>\n",
       "      <td>-0.438580</td>\n",
       "      <td>0.213846</td>\n",
       "      <td>-0.236870</td>\n",
       "      <td>-0.191679</td>\n",
       "      <td>-0.396342</td>\n",
       "      <td>0.280048</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.461572</td>\n",
       "      <td>0.124425</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7357</th>\n",
       "      <td>0.686651</td>\n",
       "      <td>0.742548</td>\n",
       "      <td>0.803792</td>\n",
       "      <td>0.052534</td>\n",
       "      <td>-0.030109</td>\n",
       "      <td>-0.544782</td>\n",
       "      <td>-0.652504</td>\n",
       "      <td>0.729326</td>\n",
       "      <td>0.677659</td>\n",
       "      <td>-0.694226</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.526258</td>\n",
       "      <td>0.657663</td>\n",
       "      <td>-0.404594</td>\n",
       "      <td>0.319675</td>\n",
       "      <td>0.346116</td>\n",
       "      <td>0.678366</td>\n",
       "      <td>-0.368306</td>\n",
       "      <td>-0.461572</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>-0.160224</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7358</th>\n",
       "      <td>-0.191131</td>\n",
       "      <td>-0.133403</td>\n",
       "      <td>-0.062469</td>\n",
       "      <td>0.197024</td>\n",
       "      <td>0.114066</td>\n",
       "      <td>0.204769</td>\n",
       "      <td>0.097779</td>\n",
       "      <td>0.038191</td>\n",
       "      <td>-0.179110</td>\n",
       "      <td>0.116280</td>\n",
       "      <td>...</td>\n",
       "      <td>0.081002</td>\n",
       "      <td>-0.042333</td>\n",
       "      <td>0.322270</td>\n",
       "      <td>0.096466</td>\n",
       "      <td>0.079304</td>\n",
       "      <td>-0.111052</td>\n",
       "      <td>-0.032168</td>\n",
       "      <td>0.124425</td>\n",
       "      <td>-0.160224</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7359 rows × 7359 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             7         8        19        20        27        30        33  \\\n",
       "0     1.000000  0.603307  0.617355  0.067536 -0.152953 -0.580836 -0.638142   \n",
       "1     0.603307  1.000000  0.737379  0.083168  0.016714 -0.508527 -0.633081   \n",
       "2     0.617355  0.737379  1.000000  0.149760 -0.043953 -0.636658 -0.691886   \n",
       "3     0.067536  0.083168  0.149760  1.000000 -0.026165 -0.048954 -0.126808   \n",
       "4    -0.152953  0.016714 -0.043953 -0.026165  1.000000  0.191237  0.105725   \n",
       "...        ...       ...       ...       ...       ...       ...       ...   \n",
       "7354  0.473162  0.657190  0.769484  0.037556  0.075790 -0.516658 -0.582472   \n",
       "7355 -0.283920 -0.368799 -0.389583 -0.132367 -0.063227  0.286439  0.371946   \n",
       "7356 -0.468367 -0.403331 -0.468688  0.026499  0.052689  0.543811  0.491412   \n",
       "7357  0.686651  0.742548  0.803792  0.052534 -0.030109 -0.544782 -0.652504   \n",
       "7358 -0.191131 -0.133403 -0.062469  0.197024  0.114066  0.204769  0.097779   \n",
       "\n",
       "            38        40        49  ...     44212     44218     44219  \\\n",
       "0     0.579787  0.688214 -0.597067  ... -0.495718  0.561550 -0.420209   \n",
       "1     0.682938  0.585571 -0.633340  ... -0.499109  0.627078 -0.332796   \n",
       "2     0.812817  0.686925 -0.688775  ... -0.535990  0.765954 -0.308007   \n",
       "3     0.224780  0.059721 -0.075750  ... -0.085676  0.127682  0.022826   \n",
       "4    -0.126072 -0.180447  0.100474  ... -0.023838 -0.152780  0.287822   \n",
       "...        ...       ...       ...  ...       ...       ...       ...   \n",
       "7354  0.677102  0.580988 -0.595140  ... -0.461170  0.664644 -0.245511   \n",
       "7355 -0.387424 -0.297418  0.385382  ...  0.310319 -0.332960  0.091233   \n",
       "7356 -0.438147 -0.429403  0.433600  ...  0.429828 -0.438580  0.213846   \n",
       "7357  0.729326  0.677659 -0.694226  ... -0.526258  0.657663 -0.404594   \n",
       "7358  0.038191 -0.179110  0.116280  ...  0.081002 -0.042333  0.322270   \n",
       "\n",
       "         44221     44227     44228     44245     44250     44254     44256  \n",
       "0     0.198508  0.301550  0.473162 -0.283920 -0.468367  0.686651 -0.191131  \n",
       "1     0.378169  0.279490  0.657190 -0.368799 -0.403331  0.742548 -0.133403  \n",
       "2     0.382220  0.363331  0.769484 -0.389583 -0.468688  0.803792 -0.062469  \n",
       "3     0.020554  0.131990  0.037556 -0.132367  0.026499  0.052534  0.197024  \n",
       "4     0.201723 -0.134249  0.075790 -0.063227  0.052689 -0.030109  0.114066  \n",
       "...        ...       ...       ...       ...       ...       ...       ...  \n",
       "7354  0.377184  0.226978  1.000000 -0.322122 -0.396342  0.678366 -0.111052  \n",
       "7355 -0.218509 -0.210024 -0.322122  1.000000  0.280048 -0.368306 -0.032168  \n",
       "7356 -0.236870 -0.191679 -0.396342  0.280048  1.000000 -0.461572  0.124425  \n",
       "7357  0.319675  0.346116  0.678366 -0.368306 -0.461572  1.000000 -0.160224  \n",
       "7358  0.096466  0.079304 -0.111052 -0.032168  0.124425 -0.160224  1.000000  \n",
       "\n",
       "[7359 rows x 7359 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cutoff = 100\n",
    "corr_meth = 'pearson'\n",
    "corr_csv = '{}_gene_correlation_{}.csv'.format(corr_meth,cutoff)\n",
    "\n",
    "corr = pd.read_csv(src + corr_csv)\n",
    "corr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Retrieve the metadata corresponding to the highly varying genes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "11 (7359, 5)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gene</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>feature_type</th>\n",
       "      <th>position_left</th>\n",
       "      <th>position_right</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Zm00001d033979</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>279975978</td>\n",
       "      <td>279978430</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Zm00001d018479</td>\n",
       "      <td>5</td>\n",
       "      <td>gene</td>\n",
       "      <td>221521585</td>\n",
       "      <td>221525322</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Zm00001d016606</td>\n",
       "      <td>5</td>\n",
       "      <td>gene</td>\n",
       "      <td>169941759</td>\n",
       "      <td>169949748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Zm00001d047989</td>\n",
       "      <td>9</td>\n",
       "      <td>gene</td>\n",
       "      <td>147322756</td>\n",
       "      <td>147334539</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Zm00001d009640</td>\n",
       "      <td>8</td>\n",
       "      <td>gene</td>\n",
       "      <td>74045694</td>\n",
       "      <td>74049815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44228</th>\n",
       "      <td>Zm00001d020669</td>\n",
       "      <td>7</td>\n",
       "      <td>gene</td>\n",
       "      <td>127426893</td>\n",
       "      <td>127431481</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44245</th>\n",
       "      <td>Zm00001d006882</td>\n",
       "      <td>2</td>\n",
       "      <td>gene</td>\n",
       "      <td>218478424</td>\n",
       "      <td>218478715</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44250</th>\n",
       "      <td>Zm00001d002347</td>\n",
       "      <td>2</td>\n",
       "      <td>gene</td>\n",
       "      <td>10657876</td>\n",
       "      <td>10658439</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44254</th>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44256</th>\n",
       "      <td>Zm00001d032828</td>\n",
       "      <td>1</td>\n",
       "      <td>gene</td>\n",
       "      <td>239455289</td>\n",
       "      <td>239464371</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7359 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 gene  chromosome feature_type  position_left  position_right\n",
       "7      Zm00001d033979           1         gene      279975978       279978430\n",
       "8      Zm00001d018479           5         gene      221521585       221525322\n",
       "19     Zm00001d016606           5         gene      169941759       169949748\n",
       "20     Zm00001d047989           9         gene      147322756       147334539\n",
       "27     Zm00001d009640           8         gene       74045694        74049815\n",
       "...               ...         ...          ...            ...             ...\n",
       "44228  Zm00001d020669           7         gene      127426893       127431481\n",
       "44245  Zm00001d006882           2         gene      218478424       218478715\n",
       "44250  Zm00001d002347           2         gene       10657876        10658439\n",
       "44254  Zm00001d028653           1         gene       41833737        41838204\n",
       "44256  Zm00001d032828           1         gene      239455289       239464371\n",
       "\n",
       "[7359 rows x 5 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "meta = gene_info.iloc[corr.columns]\n",
    "foo = meta.chromosome.unique()\n",
    "print(len(foo), meta.shape)\n",
    "meta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d047861</td>\n",
       "      <td>-15.244080</td>\n",
       "      <td>4.703940e-47</td>\n",
       "      <td>8.784256e-43</td>\n",
       "      <td>-0.012608</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d004808</td>\n",
       "      <td>-10.441780</td>\n",
       "      <td>3.181453e-24</td>\n",
       "      <td>2.339655e-20</td>\n",
       "      <td>-3.733139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d031538</td>\n",
       "      <td>-10.165090</td>\n",
       "      <td>4.220149e-23</td>\n",
       "      <td>2.880657e-19</td>\n",
       "      <td>-12.332500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d008253</td>\n",
       "      <td>-10.165090</td>\n",
       "      <td>4.220149e-23</td>\n",
       "      <td>2.880657e-19</td>\n",
       "      <td>-13.845900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1</td>\n",
       "      <td>44288</td>\n",
       "      <td>49837</td>\n",
       "      <td>Zm00001d027230</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>44306</td>\n",
       "      <td>44307</td>\n",
       "      <td>rs1_44306</td>\n",
       "      <td>Zm00001d032862</td>\n",
       "      <td>-10.165090</td>\n",
       "      <td>4.220149e-23</td>\n",
       "      <td>2.880657e-19</td>\n",
       "      <td>-0.000578</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196037</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d001752</td>\n",
       "      <td>-7.556017</td>\n",
       "      <td>9.846873e-14</td>\n",
       "      <td>3.089405e-10</td>\n",
       "      <td>-0.006062</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196038</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d001652</td>\n",
       "      <td>-7.371548</td>\n",
       "      <td>3.694633e-13</td>\n",
       "      <td>1.089526e-09</td>\n",
       "      <td>-0.042778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196039</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d045638</td>\n",
       "      <td>-7.322923</td>\n",
       "      <td>5.210988e-13</td>\n",
       "      <td>1.501189e-09</td>\n",
       "      <td>-6.382181</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196040</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d031869</td>\n",
       "      <td>-7.303307</td>\n",
       "      <td>5.983143e-13</td>\n",
       "      <td>1.712528e-09</td>\n",
       "      <td>-0.196164</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17196041</th>\n",
       "      <td>chr9</td>\n",
       "      <td>159667997</td>\n",
       "      <td>159691083</td>\n",
       "      <td>Zm00001d048577</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>159690349</td>\n",
       "      <td>159690350</td>\n",
       "      <td>rs9_159690349</td>\n",
       "      <td>Zm00001d026009</td>\n",
       "      <td>-7.003921</td>\n",
       "      <td>4.738112e-12</td>\n",
       "      <td>1.214845e-08</td>\n",
       "      <td>-0.015592</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>17196042 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "0              chr1         44288        49837  Zm00001d027230         gene   \n",
       "1              chr1         44288        49837  Zm00001d027230         gene   \n",
       "2              chr1         44288        49837  Zm00001d027230         gene   \n",
       "3              chr1         44288        49837  Zm00001d027230         gene   \n",
       "4              chr1         44288        49837  Zm00001d027230         gene   \n",
       "...             ...           ...          ...             ...          ...   \n",
       "17196037       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196038       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196039       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196040       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "17196041       chr9     159667997    159691083  Zm00001d048577         gene   \n",
       "\n",
       "         SNP_Chr  SNP_Start   SNP_Stop         SNP_ID Associated_Gene_ID  \\\n",
       "0           chr1      44306      44307      rs1_44306     Zm00001d047861   \n",
       "1           chr1      44306      44307      rs1_44306     Zm00001d004808   \n",
       "2           chr1      44306      44307      rs1_44306     Zm00001d031538   \n",
       "3           chr1      44306      44307      rs1_44306     Zm00001d008253   \n",
       "4           chr1      44306      44307      rs1_44306     Zm00001d032862   \n",
       "...          ...        ...        ...            ...                ...   \n",
       "17196037    chr9  159690349  159690350  rs9_159690349     Zm00001d001752   \n",
       "17196038    chr9  159690349  159690350  rs9_159690349     Zm00001d001652   \n",
       "17196039    chr9  159690349  159690350  rs9_159690349     Zm00001d045638   \n",
       "17196040    chr9  159690349  159690350  rs9_159690349     Zm00001d031869   \n",
       "17196041    chr9  159690349  159690350  rs9_159690349     Zm00001d026009   \n",
       "\n",
       "          Statistic          PVal           FDR       Beta  \n",
       "0        -15.244080  4.703940e-47  8.784256e-43  -0.012608  \n",
       "1        -10.441780  3.181453e-24  2.339655e-20  -3.733139  \n",
       "2        -10.165090  4.220149e-23  2.880657e-19 -12.332500  \n",
       "3        -10.165090  4.220149e-23  2.880657e-19 -13.845900  \n",
       "4        -10.165090  4.220149e-23  2.880657e-19  -0.000578  \n",
       "...             ...           ...           ...        ...  \n",
       "17196037  -7.556017  9.846873e-14  3.089405e-10  -0.006062  \n",
       "17196038  -7.371548  3.694633e-13  1.089526e-09  -0.042778  \n",
       "17196039  -7.322923  5.210988e-13  1.501189e-09  -6.382181  \n",
       "17196040  -7.303307  5.983143e-13  1.712528e-09  -0.196164  \n",
       "17196041  -7.003921  4.738112e-12  1.214845e-08  -0.015592  \n",
       "\n",
       "[17196042 rows x 14 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "eqtl = pd.read_table(eqtl_file)\n",
    "eqtl"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Making subsets of the eQTL data\n",
    "\n",
    "We subsample the eQTL matrix above. \n",
    "\n",
    "+ Consider only the eQTLs where the SNP is associated to one of the highly varying genes. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46166</th>\n",
       "      <td>chr1</td>\n",
       "      <td>3095307</td>\n",
       "      <td>3095604</td>\n",
       "      <td>Zm00001d027332</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>3095600</td>\n",
       "      <td>3095601</td>\n",
       "      <td>rs1_3095600</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89558</th>\n",
       "      <td>chr1</td>\n",
       "      <td>4494974</td>\n",
       "      <td>4499451</td>\n",
       "      <td>Zm00001d027403</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4495369</td>\n",
       "      <td>4495370</td>\n",
       "      <td>rs1_4495369</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95346</th>\n",
       "      <td>chr1</td>\n",
       "      <td>4891940</td>\n",
       "      <td>4894990</td>\n",
       "      <td>Zm00001d027419</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>4892252</td>\n",
       "      <td>4892253</td>\n",
       "      <td>rs1_4892252</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122615</th>\n",
       "      <td>chr1</td>\n",
       "      <td>5819751</td>\n",
       "      <td>5825029</td>\n",
       "      <td>Zm00001d027462</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>5824650</td>\n",
       "      <td>5824651</td>\n",
       "      <td>rs1_5824650</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142221</th>\n",
       "      <td>chr1</td>\n",
       "      <td>6407152</td>\n",
       "      <td>6409815</td>\n",
       "      <td>Zm00001d027488</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>6409375</td>\n",
       "      <td>6409376</td>\n",
       "      <td>rs1_6409375</td>\n",
       "      <td>Zm00001d032825</td>\n",
       "      <td>-6.583017</td>\n",
       "      <td>7.653515e-11</td>\n",
       "      <td>1.648439e-07</td>\n",
       "      <td>-618.7353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13559234</th>\n",
       "      <td>chr7</td>\n",
       "      <td>112328679</td>\n",
       "      <td>112337673</td>\n",
       "      <td>Zm00001d020414</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr7</td>\n",
       "      <td>112331952</td>\n",
       "      <td>112331953</td>\n",
       "      <td>rs7_112331952</td>\n",
       "      <td>Zm00001d002364</td>\n",
       "      <td>-8.035461</td>\n",
       "      <td>2.782536e-15</td>\n",
       "      <td>1.027720e-11</td>\n",
       "      <td>-141.7520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13788870</th>\n",
       "      <td>chr7</td>\n",
       "      <td>145562442</td>\n",
       "      <td>145567534</td>\n",
       "      <td>Zm00001d021196</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr7</td>\n",
       "      <td>145562868</td>\n",
       "      <td>145562869</td>\n",
       "      <td>rs7_145562868</td>\n",
       "      <td>Zm00001d002364</td>\n",
       "      <td>-8.035461</td>\n",
       "      <td>2.782536e-15</td>\n",
       "      <td>1.027720e-11</td>\n",
       "      <td>-141.7520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14746701</th>\n",
       "      <td>chr8</td>\n",
       "      <td>23386819</td>\n",
       "      <td>23390680</td>\n",
       "      <td>Zm00001d008871</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr8</td>\n",
       "      <td>23387177</td>\n",
       "      <td>23387178</td>\n",
       "      <td>rs8_23387177</td>\n",
       "      <td>Zm00001d002364</td>\n",
       "      <td>-8.035746</td>\n",
       "      <td>2.776497e-15</td>\n",
       "      <td>1.025549e-11</td>\n",
       "      <td>-141.7636</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15677638</th>\n",
       "      <td>chr8</td>\n",
       "      <td>169573762</td>\n",
       "      <td>169577154</td>\n",
       "      <td>Zm00001d012177</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr8</td>\n",
       "      <td>169575896</td>\n",
       "      <td>169575897</td>\n",
       "      <td>rs8_169575896</td>\n",
       "      <td>Zm00001d002364</td>\n",
       "      <td>-8.035461</td>\n",
       "      <td>2.782536e-15</td>\n",
       "      <td>1.027720e-11</td>\n",
       "      <td>-141.7520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16302810</th>\n",
       "      <td>chr9</td>\n",
       "      <td>38914712</td>\n",
       "      <td>38917486</td>\n",
       "      <td>Zm00001d045784</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr9</td>\n",
       "      <td>38917384</td>\n",
       "      <td>38917385</td>\n",
       "      <td>rs9_38917384</td>\n",
       "      <td>Zm00001d002364</td>\n",
       "      <td>-8.035461</td>\n",
       "      <td>2.782536e-15</td>\n",
       "      <td>1.027720e-11</td>\n",
       "      <td>-141.7520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>469581 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "46166          chr1       3095307      3095604  Zm00001d027332         gene   \n",
       "89558          chr1       4494974      4499451  Zm00001d027403         gene   \n",
       "95346          chr1       4891940      4894990  Zm00001d027419         gene   \n",
       "122615         chr1       5819751      5825029  Zm00001d027462         gene   \n",
       "142221         chr1       6407152      6409815  Zm00001d027488         gene   \n",
       "...             ...           ...          ...             ...          ...   \n",
       "13559234       chr7     112328679    112337673  Zm00001d020414         gene   \n",
       "13788870       chr7     145562442    145567534  Zm00001d021196         gene   \n",
       "14746701       chr8      23386819     23390680  Zm00001d008871         gene   \n",
       "15677638       chr8     169573762    169577154  Zm00001d012177         gene   \n",
       "16302810       chr9      38914712     38917486  Zm00001d045784         gene   \n",
       "\n",
       "         SNP_Chr  SNP_Start   SNP_Stop         SNP_ID Associated_Gene_ID  \\\n",
       "46166       chr1    3095600    3095601    rs1_3095600     Zm00001d032825   \n",
       "89558       chr1    4495369    4495370    rs1_4495369     Zm00001d032825   \n",
       "95346       chr1    4892252    4892253    rs1_4892252     Zm00001d032825   \n",
       "122615      chr1    5824650    5824651    rs1_5824650     Zm00001d032825   \n",
       "142221      chr1    6409375    6409376    rs1_6409375     Zm00001d032825   \n",
       "...          ...        ...        ...            ...                ...   \n",
       "13559234    chr7  112331952  112331953  rs7_112331952     Zm00001d002364   \n",
       "13788870    chr7  145562868  145562869  rs7_145562868     Zm00001d002364   \n",
       "14746701    chr8   23387177   23387178   rs8_23387177     Zm00001d002364   \n",
       "15677638    chr8  169575896  169575897  rs8_169575896     Zm00001d002364   \n",
       "16302810    chr9   38917384   38917385   rs9_38917384     Zm00001d002364   \n",
       "\n",
       "          Statistic          PVal           FDR      Beta  \n",
       "46166     -6.583017  7.653515e-11  1.648439e-07 -618.7353  \n",
       "89558     -6.583017  7.653515e-11  1.648439e-07 -618.7353  \n",
       "95346     -6.583017  7.653515e-11  1.648439e-07 -618.7353  \n",
       "122615    -6.583017  7.653515e-11  1.648439e-07 -618.7353  \n",
       "142221    -6.583017  7.653515e-11  1.648439e-07 -618.7353  \n",
       "...             ...           ...           ...       ...  \n",
       "13559234  -8.035461  2.782536e-15  1.027720e-11 -141.7520  \n",
       "13788870  -8.035461  2.782536e-15  1.027720e-11 -141.7520  \n",
       "14746701  -8.035746  2.776497e-15  1.025549e-11 -141.7636  \n",
       "15677638  -8.035461  2.782536e-15  1.027720e-11 -141.7520  \n",
       "16302810  -8.035461  2.782536e-15  1.027720e-11 -141.7520  \n",
       "\n",
       "[469581 rows x 14 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "related = pd.DataFrame()\n",
    "\n",
    "for i in range(meta.shape[0]):\n",
    "    related = related.append(eqtl[ eqtl['Associated_Gene_ID'] == meta.iloc[i]['gene']], sort=False)\n",
    "\n",
    "related"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1642 genes out of 7359 contain associated SNPs\n"
     ]
    }
   ],
   "source": [
    "print(len(related.Associated_Gene_ID.unique()), 'genes out of', len(meta.gene.unique()),\n",
    "      'contain associated SNPs')\n",
    "related.drop(columns=['SNP_Chr','SNP_Stop','R_Gene_Class']).to_csv(dst+'eqtl_related_{}.csv'.format(cutoff),\n",
    "                                                                  index = True, index_label='Original_Index')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>R_Gene_Chr</th>\n",
       "      <th>R_Gene_Start</th>\n",
       "      <th>R_Gene_Stop</th>\n",
       "      <th>R_Gene_ID</th>\n",
       "      <th>R_Gene_Class</th>\n",
       "      <th>SNP_Chr</th>\n",
       "      <th>SNP_Start</th>\n",
       "      <th>SNP_Stop</th>\n",
       "      <th>SNP_ID</th>\n",
       "      <th>Associated_Gene_ID</th>\n",
       "      <th>Statistic</th>\n",
       "      <th>PVal</th>\n",
       "      <th>FDR</th>\n",
       "      <th>Beta</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>10417261</th>\n",
       "      <td>chr5</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr5</td>\n",
       "      <td>69975582</td>\n",
       "      <td>69975583</td>\n",
       "      <td>rs5_69975582</td>\n",
       "      <td>Zm00001d040202</td>\n",
       "      <td>-7.155506</td>\n",
       "      <td>1.677400e-12</td>\n",
       "      <td>4.564204e-09</td>\n",
       "      <td>-580.0772</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417335</th>\n",
       "      <td>chr5</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr5</td>\n",
       "      <td>69975588</td>\n",
       "      <td>69975589</td>\n",
       "      <td>rs5_69975588</td>\n",
       "      <td>Zm00001d040202</td>\n",
       "      <td>-7.563082</td>\n",
       "      <td>9.355389e-14</td>\n",
       "      <td>2.941734e-10</td>\n",
       "      <td>-611.2713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10417910</th>\n",
       "      <td>chr5</td>\n",
       "      <td>69975364</td>\n",
       "      <td>69978349</td>\n",
       "      <td>Zm00001d014957</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr5</td>\n",
       "      <td>69978178</td>\n",
       "      <td>69978179</td>\n",
       "      <td>rs5_69978178</td>\n",
       "      <td>Zm00001d040202</td>\n",
       "      <td>-7.563082</td>\n",
       "      <td>9.355389e-14</td>\n",
       "      <td>2.941734e-10</td>\n",
       "      <td>-611.2713</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12014337</th>\n",
       "      <td>chr6</td>\n",
       "      <td>92808874</td>\n",
       "      <td>92811320</td>\n",
       "      <td>Zm00001d036567</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr6</td>\n",
       "      <td>92811304</td>\n",
       "      <td>92811305</td>\n",
       "      <td>rs6_92811304</td>\n",
       "      <td>Zm00001d001139</td>\n",
       "      <td>-8.917276</td>\n",
       "      <td>2.446043e-18</td>\n",
       "      <td>1.199951e-14</td>\n",
       "      <td>-1262.5240</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14255497</th>\n",
       "      <td>chr7</td>\n",
       "      <td>173953536</td>\n",
       "      <td>173954457</td>\n",
       "      <td>Zm00001d022264</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr7</td>\n",
       "      <td>173954253</td>\n",
       "      <td>173954254</td>\n",
       "      <td>rs7_173954253</td>\n",
       "      <td>Zm00001d044495</td>\n",
       "      <td>-8.451948</td>\n",
       "      <td>1.081581e-16</td>\n",
       "      <td>4.604894e-13</td>\n",
       "      <td>-823.7607</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616952</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41836775</td>\n",
       "      <td>41836776</td>\n",
       "      <td>rs1_41836775</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.789329</td>\n",
       "      <td>7.050030e-18</td>\n",
       "      <td>3.306002e-14</td>\n",
       "      <td>-288.7677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>616999</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41836824</td>\n",
       "      <td>41836825</td>\n",
       "      <td>rs1_41836824</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.789329</td>\n",
       "      <td>7.050030e-18</td>\n",
       "      <td>3.306002e-14</td>\n",
       "      <td>-288.7677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617046</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41836963</td>\n",
       "      <td>41836964</td>\n",
       "      <td>rs1_41836963</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.789329</td>\n",
       "      <td>7.050030e-18</td>\n",
       "      <td>3.306002e-14</td>\n",
       "      <td>-288.7677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617075</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837097</td>\n",
       "      <td>41837098</td>\n",
       "      <td>rs1_41837097</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-8.439968</td>\n",
       "      <td>1.189764e-16</td>\n",
       "      <td>5.001608e-13</td>\n",
       "      <td>-268.1414</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>617171</th>\n",
       "      <td>chr1</td>\n",
       "      <td>41833737</td>\n",
       "      <td>41838204</td>\n",
       "      <td>Zm00001d028653</td>\n",
       "      <td>gene</td>\n",
       "      <td>chr1</td>\n",
       "      <td>41837500</td>\n",
       "      <td>41837501</td>\n",
       "      <td>rs1_41837500</td>\n",
       "      <td>zma-MIR482</td>\n",
       "      <td>-9.638586</td>\n",
       "      <td>4.965583e-21</td>\n",
       "      <td>2.961445e-17</td>\n",
       "      <td>-341.2709</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1007 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         R_Gene_Chr  R_Gene_Start  R_Gene_Stop       R_Gene_ID R_Gene_Class  \\\n",
       "10417261       chr5      69975364     69978349  Zm00001d014957         gene   \n",
       "10417335       chr5      69975364     69978349  Zm00001d014957         gene   \n",
       "10417910       chr5      69975364     69978349  Zm00001d014957         gene   \n",
       "12014337       chr6      92808874     92811320  Zm00001d036567         gene   \n",
       "14255497       chr7     173953536    173954457  Zm00001d022264         gene   \n",
       "...             ...           ...          ...             ...          ...   \n",
       "616952         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "616999         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617046         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617075         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "617171         chr1      41833737     41838204  Zm00001d028653         gene   \n",
       "\n",
       "         SNP_Chr  SNP_Start   SNP_Stop         SNP_ID Associated_Gene_ID  \\\n",
       "10417261    chr5   69975582   69975583   rs5_69975582     Zm00001d040202   \n",
       "10417335    chr5   69975588   69975589   rs5_69975588     Zm00001d040202   \n",
       "10417910    chr5   69978178   69978179   rs5_69978178     Zm00001d040202   \n",
       "12014337    chr6   92811304   92811305   rs6_92811304     Zm00001d001139   \n",
       "14255497    chr7  173954253  173954254  rs7_173954253     Zm00001d044495   \n",
       "...          ...        ...        ...            ...                ...   \n",
       "616952      chr1   41836775   41836776   rs1_41836775         zma-MIR482   \n",
       "616999      chr1   41836824   41836825   rs1_41836824         zma-MIR482   \n",
       "617046      chr1   41836963   41836964   rs1_41836963         zma-MIR482   \n",
       "617075      chr1   41837097   41837098   rs1_41837097         zma-MIR482   \n",
       "617171      chr1   41837500   41837501   rs1_41837500         zma-MIR482   \n",
       "\n",
       "          Statistic          PVal           FDR       Beta  \n",
       "10417261  -7.155506  1.677400e-12  4.564204e-09  -580.0772  \n",
       "10417335  -7.563082  9.355389e-14  2.941734e-10  -611.2713  \n",
       "10417910  -7.563082  9.355389e-14  2.941734e-10  -611.2713  \n",
       "12014337  -8.917276  2.446043e-18  1.199951e-14 -1262.5240  \n",
       "14255497  -8.451948  1.081581e-16  4.604894e-13  -823.7607  \n",
       "...             ...           ...           ...        ...  \n",
       "616952    -8.789329  7.050030e-18  3.306002e-14  -288.7677  \n",
       "616999    -8.789329  7.050030e-18  3.306002e-14  -288.7677  \n",
       "617046    -8.789329  7.050030e-18  3.306002e-14  -288.7677  \n",
       "617075    -8.439968  1.189764e-16  5.001608e-13  -268.1414  \n",
       "617171    -9.638586  4.965583e-21  2.961445e-17  -341.2709  \n",
       "\n",
       "[1007 rows x 14 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "locrel = pd.concat([located, related], axis=1, join='inner')\n",
    "locrel = locrel.iloc[:, :located.shape[1]]\n",
    "locrel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "92 genes out of 648 contain associated SNPs\n"
     ]
    }
   ],
   "source": [
    "print(len(locrel.Associated_Gene_ID.unique()), 'genes out of', len(meta.gene.unique()),\n",
    "      'contain associated SNPs')\n",
    "locrel.drop(columns=['SNP_Chr','SNP_Stop','R_Gene_Class']).to_csv(dst+'eqtl_locrel_{}.csv'.format(cutoff),\n",
    "                                                                  index = True, index_label='Original_Index')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
